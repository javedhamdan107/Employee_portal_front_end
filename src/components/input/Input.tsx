import './styles.css'
type InputPropTypes={
    value:string;
    onChange :(e) => void;
    label :string;
    type: 'text'|'password'

}
const Input :React.FC <InputPropTypes> = ({label,value,type,onChange})=>{


return(
    <div>
        <div className='relative'>
        <input className='input'type={type} value={value} onChange={onChange} />
        <label className={value.length>0&&`activeLabel`}>{label}</label>
        </div>

    </div>
)
}

export default Input;