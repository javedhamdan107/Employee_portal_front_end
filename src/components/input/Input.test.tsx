
import { fireEvent, render, screen } from "@testing-library/react";
import Input, { InputPropTypes } from "./Input";
describe("If Button works properly", () => {
  test("If label rendered correctly", () => {
    const onChange = jest.fn();
    const InputProps: InputPropTypes = {
      label: "username",
      value: "",
      type: "text",
      onChange
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId("label-test");

    expect(element.innerHTML).toBe("username");
  });

  test("If type text rendered correctly", () => {
    const onChange = jest.fn();
    const InputProps: InputPropTypes = {
      label: "username",
      value: "",
      type: "text",
      onChange
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId("input-test");

    expect(element.getAttribute("type")).toBe("text");
  });

  test("If type password rendered correctly", () => {
    const onChange = jest.fn();
    const InputProps: InputPropTypes = {
      label: "username",
      value: "",
      type: "password",
      onChange
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId("input-test");

    expect(element.getAttribute("type")).toBe("password");
  });
  test("If Snapshot exists correctly", () => {
    const onChange = jest.fn();
    const InputProps: InputPropTypes = {
      label: "username",
      value: "",
      type: "password",
      onChange
    };

    const { asFragment } = render(<Input {...InputProps} />);

    expect(asFragment).toMatchSnapshot();
  });
  test("If onChange is triggered", () => {
    const onChange = jest.fn();
    const InputProps: InputPropTypes = {
      label: "username",
      value: "",
      type: "password",
      onChange
    };

    render(<Input {...InputProps} />);
    const element = screen.getByTestId("input-test");

    fireEvent.change(element, { target: { value: "new_value" } });
    expect(onChange).toBeCalled();
  });
});
