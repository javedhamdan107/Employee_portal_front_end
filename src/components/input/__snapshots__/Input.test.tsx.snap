// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If Button works properly If Snapshot exists correctly 1`] = `[Function]`;
